Assets/Scripts/GUI/MultiplayerPanel.cs

```diff
@@ -32,18 +34,30 @@ public string RoomName
             set
             {
                 data.roomName = value;
-                UpdateRoomNumberDisplay(); 
+                UpdateRoomNumberDisplay();
+                UpdateRoomExistenceMessage(); 
             }
         }
 
         private RoomCreateData data = new RoomCreateData
         {
-            roomName = GenerateRandomRoomName(),
+            roomName = GenerateUniqueRoomName(),
             @private = false,
             maxPlayers = 4,
             voiceDisabled = false
         };
 
+        private static string GenerateUniqueRoomName()
+        {
+            string roomName;
+            do
+            {
+                roomName = GenerateRandomRoomName();
+            } while (MultiplayerManager.m_Instance.DoesRoomNameExist(roomName));
```

Comment:
Should we have a failsafe? An infinite loop might creep in because of a bug elsewhere. I never trust while loops without a guaranteed exit condition!